### PR TITLE
46 feat parse body content length

### PR DESCRIPTION
**Pull request by sabe-0524**
content-lengthが指定されている時のbody parser
lengthを超えない範囲でbufferをbodyに入れる

## Summary by CodeRabbit

* **New Features**
  * Content-Length-based request body handling and initial support path for chunked Transfer-Encoding
  * Content-Length exposed via a public accessor for diagnostics

* **Bug Fixes**
  * Stronger validation and error reporting for Transfer-Encoding and Content-Length
  * Enforced client body size limits with new error conditions

* **Tests**
  * Added comprehensive body-parsing tests covering fragmentation, binary payloads, limits, and edge cases

* **Chores**
  * Default maximum client body size set to 1MB

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/HttpRequest.cpp [hunk 1]
```diff
@@ -74,6 +74,8 @@ void HttpRequest::clear() {
   _version.clear();
   _headers.clear();
   _body.clear();
+  _contentLength = 0;
+  _isChunked = false;
```
Comment (tnktks111) [inline src/HttpRequest.cpp]: コンストラクタでも初期化が必要？

**Review by tnktks111 — COMMENTED** [review]
comment
